### ISSUE TITLE
Disk Status is blank

### ISSUE THREAD

**Issue by warplayer** [Swizards/QuickBox#44]
Since I installed Quickbox the Disk Status pane hasn't displayed anything. I'm not really sure where to look for settings. I figure I just need to point it to the partition I'm using. (/dev/md3)

**Comment by JMSDOnline** [comment]
@warplayer, have you typed in the command ``setdisk`` from command line to initiate the quota allowance per specific users?

**Comment by warplayer** [comment]
Yes, I had a hard time getting it to go above 1.7TB. I got a bit confused when trying to do setdisk. I should be able to assign 3TB I think. When I do a df it looks like this: http://imgur.com/NC1QKNr

**Comment by JMSDOnline** [comment]
Ah, I see. You're setup is based on a /home partition layout. There are currently some issues with that. You will need to reassign quotas to measure from ``/home`` rather than``/``. Also, the diskspace plugin provided via ruTorrent will need to have some adjustments.

So, reset your quotas to read the read partition as follows:

```
quotaoff -a
service quota stop
mount -o remount /home
quotacheck -auMF vfsv1
quotaon -uv /home
service quota start
```

Then to modify the diskspace plugin it would be:
modify the action.php located at --> **/srv/rutorrent/plugins/diskspace/action.php**
You'll see ``/usr/sbin/repquota -u /`` within both lines for $total and $free. Change that to ``/usr/sbin/repquota -u /home`` on both values.

This should adjust both ruTorrent as well as QuickBox to read the proper mount.

Just a word to the wise, I haven't had time to really dig my heels in and mod the script to be dynamic. Doing this should not cause any adverse effects as you can always sort it back. I usually recommend if using a seedbox to waive the **/home** mount point and focus on **/** for quotas.

This gets especially awkward when it involves two separate mount points and/or disks that are not in raid.

I'll try to find time to sort this out.

**Comment by JMSDOnline** [comment]
Also, I think you were pointing at an example of 1.7TB. To set that you can set the value in MB and it will read it properly. 

1740GB converts to 1.7TB

**Comment by warplayer** [comment]
Thanks for all the great info. I wouldn't mind just using / instead of /home. I saw I had around 1 TB of space in / and around 1.7TB in /home, and I was trying to figure out how to just merge the two so I could have the full 2.7TB available but I couldn't figure out how. I'm not super well versed in linux. Thanks again for the help.

**Comment by sohalartz** [comment]
Today i have install QuickBox everything goes well except disk status . i also run setdisk command in terminal but server says setdisk not found .

**Comment by JMSDOnline** [comment]
@sohalartz, could you provide a few more details? 

1. What OS/Distro are you using
2. Are you using a /home primary partition
3. What are the results of ``df -h`` via ssh
4. Are you running the ``setdisk`` command as the master and/or root account
5. Did you install the script as root

Thanks!

**Comment by warplayer** [comment]
Okay so I finally got around to trying the steps you outlined above and I got an error.

[root@warbox]:(11.7kb)/# quotaon -uv /home
quotaon: Mountpoint (or device) /home not found or has no quota enabled.

I think I am just fundamentally not understanding user disk quotas. My "df -h" is still the same from the screen shot above. Starting from the beginning, how can I set my seedbox to use the max disk space available to me? As far as I can see I should be able to use just under 3TB but right now im limited to 1.7TB. 

Thanks again for the help.

**Comment by JMSDOnline** [comment]
Hey @warplayer,

This is actually something you would need to setup with your hosting provider. What is happening is it looks like your system is not raid enabled. You have two disks, in order to utilize them as essentially one, you'll need a raid setup.. i.e; RAID 0 is a fairly standard choice.

If you're ok with starting from scratch, I would request a reinstall of this server and request it be done in ``RAID 0`` with the primary partition (and all space on) being ``/``

If you get it reinstalled or need any additional quick support, feel free to contact me via jmsolodesigns[at]gmail[dot]com

I will gladly throw some assistance your way given I have the time to do so.

**Comment by sohalartz** [comment]
@JMSDOnline i'm using Ubuntu server 14.04 x86_64 
output for df -h 
Filesystem      Size  Used Avail Use% Mounted on
udev            3.9G   12K  3.9G   1% /dev
tmpfs           799M  340K  799M   1% /run
/dev/vda1        79G   14G   63G  18% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
none            5.0M     0  5.0M   0% /run/lock
none            3.9G   72K  3.9G   1% /run/shm
none            100M   12K  100M   1% /run/user

i run setdisk from root user and also install script from root

**Comment by JMSDOnline** [comment]
@sohalartz, interesting... thanks for taking the time to send me some feedback. By chance, what is returned with the command ``quotaon -uv /``?

We can use a standalone ``setdisk`` script - I just want to make certain that quotas are good before attempting a fix.

**Comment by sohalartz** [comment]
@JMSDOnline 
output for quotaon -uv /
quotaon: Mountpoint (or device) / not found or has no quota enabled.

**Comment by sohalartz** [comment]
@JMSDOnline 
[quickbox.txt](https://github.com/Swizards/QuickBox/files/222235/quickbox.txt)
i made new installation this time setdisk command working but in admin panel still disk status is empty . log file attached .

**Comment by JMSDOnline** [comment]
@sohalartz, Is the result for ``quotaon -uv /`` still the same?

**Comment by sohalartz** [comment]
@JMSDOnline yes same as above

**Comment by JMSDOnline** [comment]
@sohalartz, could you show me the return of ``cat /etc/fstab``

**Comment by sohalartz** [comment]
LABEL=cloudimg-rootfs   /        ext4   defaults        0 0

**Comment by sohalartz** [comment]
@JMSDOnline i tried original Novik Diskspace plugin it works in rutorrent ui but not in admin panel so i revert back again to your modified plugin.

**Comment by JMSDOnline** [comment]
@warplayer, since /home is your primary mount, you may want to look into my post and patch here #58 . 

@sohalartz, you may need to actually rebuild your quotas as it looks like they never got set via the script. I am not too certain what your environment for the server is so it may be how the provider has delivered it... and it just so happens to be completely new to me.

Your step may involve a little more muscle as follows:
```
quotaoff -a
service quota stop
sed -i 's/errors=remount-ro/usrquota,errors=remount-ro/g' /etc/fstab
mount -o remount / || mount -o remount /home
quotacheck -auMF vfsv1
quotaon -uv /
service quota start
```

The rest of the steps listed at #58 in regards to the diskspace plugin may not be necessary in your case... but they will certainly be needed by @warplayer.

**Comment by sohalartz** [comment]
@JMSDOnline i modified `/etc/fstab`

`LABEL=cloudimg-rootfs   /        ext4   defaults      0 0`
to
`LABEL=cloudimg-rootfs   /        ext4   defaults,usrquota,grpquota      0 0`

then reboot

`quotaoff -a`
`service quota stop`
`mount -o remount /`
`quotacheck -auMF vfsv1`
`quotaon -uv /`
`service quota start`
run `setdisk` command to set quota

**Comment by sohalartz** [comment]
@JMSDOnline disk status is working but need to update stats manually from terminal by running 
`quotacheck -auMF vfsv1`

**Comment by JMSDOnline** [comment]
@sohalartz, yeah, your grpquota is more than likely going to be the culprit of that issue. 

```
LABEL=cloudimg-rootfs / ext4    usrquota,errors=remount-ro 0 1
```

I would try to stay on track with that

**Comment by sohalartz** [comment]
@JMSDOnline Finally i find a perfect solution which is working 
i'm on kvm and quota is not enabled in virtual machine kernel so i need to enable it manually .
here is explained tutorial how to enable quota https://binaryguru.wordpress.com/2014/10/17/webmin-1-710-on-ubuntu-server-14-04-64-bit-running-on-amazon-ec2/
